### PR TITLE
OCPBUGS-20516: rapid reset CVE fix

### PR DESCRIPTION
**Pull request by everettraven**
**Description of the change:**
- Replaces k8s packages with https://github.com/openshift/kubernetes/commit/c7606e743ff1b002c11b4d72da2add10d2057a9d
  - Uses this commit because this is a commit on the `release-4.13` branch which corresponds to the 1.26.z k8s packages which are still used here on `master`
- Bumps `golang.org/x/net` to `v0.17.0`
- `go mod tidy`
- `go mod vendor`

### CODE REVIEW

**Review by oceanc80 — APPROVED** [review]

**Review by rashmigottipati — APPROVED** [review]
/lgtm

File: go.mod [hunk 1]
```diff
@@ -248,3 +248,16 @@ replace (
 	github.com/docker/distribution => github.com/docker/distribution v0.0.0-20191216044856-a8371794149d
 	github.com/mattn/go-sqlite3 => github.com/mattn/go-sqlite3 v1.10.0
 )
+
+// Replacements to resolve the rapid reset CVE. This should be removed
+// when the upstream patched versions are released and downstreamed.
+// https://access.redhat.com/security/cve/CVE-2023-44487
+// https://access.redhat.com/security/cve/CVE-2023-39325
```
Comment (joelanford) [inline go.mod:255]: Can we make downstream github issues/Jira issues to make sure we come back to this?

**Review by KeenonLee — COMMENTED** [review]
/label qe-approved

**Pull request merged**